pandas/tests/indexing/multiindex/test_loc.py

```diff
@@ -598,3 +598,15 @@ def test_getitem_loc_commutability(multiindex_year_month_day_dataframe_random_da
     result = ser[2000, 5]
     expected = df.loc[2000, 5]["A"]
     tm.assert_series_equal(result, expected)
+
+
+def test_loc_with_nan():
+    # GH: 27104
+    df = DataFrame(
+        {"col": [1, 2, 5], "ind1": ["a", "d", np.nan], "ind2": [1, 4, 5]}
+    ).set_index(["ind1", "ind2"])
```

Comment:
can you also select `df.loc['a']` as the first case in OP (which is working). can parameterize if easy.

Reply:
Parametrization would be ugly, because Index needs new definition. Added the test below.

Reply:
kk totally fine, i always ask, but like you did push back if its non-trivial or would make it less obvious.